src/conn/pool/mod.rs

```diff
@@ -220,6 +226,10 @@ impl Pool {
             inner: Arc::new(Inner {
                 close: false.into(),
                 closed: false.into(),
+                metrics: Arc::new(Metrics {
+                    connection_count: atomic::AtomicUsize::new(pool_opts.constraints().max()),
```

Comment:
The pool is lazy, so `connection_count` should start with zero i believe.

Reply:
good spot thanks, fixed